pkg/provisioner/controller.go

```diff
@@ -462,33 +462,38 @@ func (c *obcController) supportedProvisioner(provisioner string) bool {
 	return provisioner == c.provisionerName
 }
 
-// Returns the ob, configmap, and secret based on the passed-in key. Only returns non-nil
-// error if unable to get all resources. Some resources may be nil.
-func (c *obcController) getResourcesFromKey(key string) (*v1alpha1.ObjectBucket, *corev1.ConfigMap, *corev1.Secret, error) {
-
-	ob, obErr := c.objectBucketForClaimKey(key)
-	if errors.IsNotFound(obErr) {
-		log.Error(obErr, "objectBucket not found")
-		obErr = nil
-	}
-	cm, cmErr := configMapForClaimKey(key, c.clientset)
-	if errors.IsNotFound(cmErr) {
-		log.Error(cmErr, "configMap not found")
-		cmErr = nil
-	}
-	s, sErr := secretForClaimKey(key, c.clientset)
-	if errors.IsNotFound(sErr) {
-		log.Error(sErr, "secret not found")
-		sErr = nil
+// trim the errors resulting from objects not being found
+func (c *obcController) getResourcesForDeletionByKey(key string) (*v1alpha1.ObjectBucket, *corev1.ConfigMap, *corev1.Secret, []error) {
+	ob, cm, secret, errs := c.getResourcesFromKey(key)
+	for i := len(errs) - 1; i >= 0; i-- {
+		if errors.IsNotFound(errs[i]) {
```

Comment:
What about passing an _ignoreNotFoundErr_ to `getResourcesFromKey` and then this func would not be needed?

Reply:
I see your point in that it could probably reduce lines of code and I think that overloads the intent of `getResourcesFromKey`, and perhaps it's reusability.   That said, naming the function specific to deletion also misstates what it does.  What do you think about delineating abstraction between "give me everything" and "give must just what exists"?

Reply:
Yes, that abstraction works for me. +1